From 5b5578f327909a05234baa9c3e9e98fbb2d91a59 Mon Sep 17 00:00:00 2001
From: Juan Pablo Ruiz <juanpablo.ruiz@unikie.com>
Date: Fri, 20 Dec 2024 20:34:20 +0200
Subject: [PATCH 1/1] nvidia-oot: Fix build for Linux v6.12

- All .llseek to NULL
- Replaced include <asm/unaligned.h> to #include <linux/unaligned.h>
- Removed .output_poll_changed from drm_mode_config_funcs
- Removed .lastclose from drm_driver

Signed-off-by: Juan Pablo Ruiz <juanpablo.ruiz@unikie.com>
---
 drivers/bluetooth/realtek/rtk_bt.c                            | 2 +-
 drivers/bluetooth/realtek/rtk_misc.c                          | 2 +-
 drivers/gpu/drm/tegra/drm.c                                   | 4 ----
 drivers/gpu/drm/tegra/riscv.c                                 | 2 +-
 .../platform/tegra/camera/fusa-capture/capture-isp-channel.c  | 2 +-
 .../platform/tegra/camera/fusa-capture/capture-vi-channel.c   | 2 +-
 drivers/misc/nvsciipc/nvsciipc.c                              | 2 +-
 drivers/platform/tegra/rtcpu/camchar.c                        | 2 +-
 drivers/video/tegra/host/nvcsi/nvcsi-t194.c                   | 2 +-
 drivers/video/tegra/host/nvcsi/nvcsi.c                        | 2 +-
 drivers/video/tegra/host/nvdla/nvdla_ioctl.c                  | 2 +-
 drivers/video/tegra/host/pva/pva_ioctl.c                      | 2 +-
 drivers/virt/tegra/tegra_hv_pm_ctl.c                          | 2 +-
 13 files changed, 12 insertions(+), 16 deletions(-)

diff --git a/drivers/bluetooth/realtek/rtk_bt.c b/drivers/bluetooth/realtek/rtk_bt.c
index 95b7b7e4..03ce78ed 100644
--- a/drivers/bluetooth/realtek/rtk_bt.c
+++ b/drivers/bluetooth/realtek/rtk_bt.c
@@ -17,7 +17,7 @@
 #include <linux/dcache.h>
 #include <linux/reboot.h>
 #include <net/sock.h>
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 
 #include "rtk_bt.h"
 #include "rtk_misc.h"
diff --git a/drivers/bluetooth/realtek/rtk_misc.c b/drivers/bluetooth/realtek/rtk_misc.c
index 5dec1a5f..880ac655 100644
--- a/drivers/bluetooth/realtek/rtk_misc.c
+++ b/drivers/bluetooth/realtek/rtk_misc.c
@@ -17,7 +17,7 @@
 #include <linux/dcache.h>
 #include <linux/in.h>
 #include <net/sock.h>
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 
 #include <linux/interrupt.h>
 #include <linux/module.h>
diff --git a/drivers/gpu/drm/tegra/drm.c b/drivers/gpu/drm/tegra/drm.c
index 657415be..8ffa9800 100644
--- a/drivers/gpu/drm/tegra/drm.c
+++ b/drivers/gpu/drm/tegra/drm.c
@@ -61,9 +61,6 @@ static int tegra_atomic_check(struct drm_device *drm,
 
 static const struct drm_mode_config_funcs tegra_drm_mode_config_funcs = {
 	.fb_create = tegra_fb_create,
-#ifdef CONFIG_DRM_FBDEV_EMULATION
-	.output_poll_changed = drm_fb_helper_output_poll_changed,
-#endif
 	.atomic_check = tegra_atomic_check,
 	.atomic_commit = drm_atomic_helper_commit,
 };
@@ -886,7 +883,6 @@ static const struct drm_driver tegra_drm_driver = {
 			   DRIVER_ATOMIC | DRIVER_RENDER | DRIVER_SYNCOBJ,
 	.open = tegra_drm_open,
 	.postclose = tegra_drm_postclose,
-	.lastclose = drm_fb_helper_lastclose,
 
 #if defined(CONFIG_DEBUG_FS)
 	.debugfs_init = tegra_debugfs_init,
diff --git a/drivers/gpu/drm/tegra/riscv.c b/drivers/gpu/drm/tegra/riscv.c
index cee9ba2f..4af0279e 100644
--- a/drivers/gpu/drm/tegra/riscv.c
+++ b/drivers/gpu/drm/tegra/riscv.c
@@ -3,7 +3,7 @@
  * Copyright (c) 2022, NVIDIA Corporation.
  */
 
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 
 #include <linux/dev_printk.h>
 #include <linux/device.h>
diff --git a/drivers/media/platform/tegra/camera/fusa-capture/capture-isp-channel.c b/drivers/media/platform/tegra/camera/fusa-capture/capture-isp-channel.c
index b6ef7b53..4d36494f 100644
--- a/drivers/media/platform/tegra/camera/fusa-capture/capture-isp-channel.c
+++ b/drivers/media/platform/tegra/camera/fusa-capture/capture-isp-channel.c
@@ -523,7 +523,7 @@ static long isp_channel_ioctl(
 
 static const struct file_operations isp_channel_fops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = isp_channel_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = isp_channel_ioctl,
diff --git a/drivers/media/platform/tegra/camera/fusa-capture/capture-vi-channel.c b/drivers/media/platform/tegra/camera/fusa-capture/capture-vi-channel.c
index 0ab7be1d..9286e8ae 100644
--- a/drivers/media/platform/tegra/camera/fusa-capture/capture-vi-channel.c
+++ b/drivers/media/platform/tegra/camera/fusa-capture/capture-vi-channel.c
@@ -616,7 +616,7 @@ static long vi_channel_ioctl(
 
 static const struct file_operations vi_channel_fops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = vi_channel_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = vi_channel_ioctl,
diff --git a/drivers/misc/nvsciipc/nvsciipc.c b/drivers/misc/nvsciipc/nvsciipc.c
index 7ad5de41..459d3379 100644
--- a/drivers/misc/nvsciipc/nvsciipc.c
+++ b/drivers/misc/nvsciipc/nvsciipc.c
@@ -685,7 +685,7 @@ static const struct file_operations nvsciipc_fops = {
 	.open		= nvsciipc_dev_open,
 	.release		= nvsciipc_dev_release,
 	.unlocked_ioctl	= nvsciipc_dev_ioctl,
-	.llseek		= no_llseek,
+	.llseek		= NULL,
 	.read		= nvsciipc_dbg_read,
 };
 
diff --git a/drivers/platform/tegra/rtcpu/camchar.c b/drivers/platform/tegra/rtcpu/camchar.c
index c271f19d..4c9706ba 100644
--- a/drivers/platform/tegra/rtcpu/camchar.c
+++ b/drivers/platform/tegra/rtcpu/camchar.c
@@ -226,7 +226,7 @@ static const struct file_operations tegra_camchar_fops = {
 	.release = tegra_camchar_release,
 	.unlocked_ioctl = tegra_camchar_ioctl,
 	.compat_ioctl = tegra_camchar_ioctl,
-	.llseek = no_llseek,
+	.llseek = NULL,
 };
 
 static int __init tegra_camchar_init(struct tegra_ivc_driver *drv)
diff --git a/drivers/video/tegra/host/nvcsi/nvcsi-t194.c b/drivers/video/tegra/host/nvcsi/nvcsi-t194.c
index 75f6dc05..6dc95852 100644
--- a/drivers/video/tegra/host/nvcsi/nvcsi-t194.c
+++ b/drivers/video/tegra/host/nvcsi/nvcsi-t194.c
@@ -110,7 +110,7 @@ static int t194_nvcsi_release(struct inode *inode, struct file *file)
 
 const struct file_operations tegra194_nvcsi_ctrl_ops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = t194_nvcsi_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = t194_nvcsi_ioctl,
diff --git a/drivers/video/tegra/host/nvcsi/nvcsi.c b/drivers/video/tegra/host/nvcsi/nvcsi.c
index 2cc96809..c34b8020 100644
--- a/drivers/video/tegra/host/nvcsi/nvcsi.c
+++ b/drivers/video/tegra/host/nvcsi/nvcsi.c
@@ -129,7 +129,7 @@ static int nvcsi_release(struct inode *inode, struct file *file)
 
 const struct file_operations tegra_nvcsi_ctrl_ops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = nvcsi_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = nvcsi_ioctl,
diff --git a/drivers/video/tegra/host/nvdla/nvdla_ioctl.c b/drivers/video/tegra/host/nvdla/nvdla_ioctl.c
index 1ab08669..2cb9a2f7 100644
--- a/drivers/video/tegra/host/nvdla/nvdla_ioctl.c
+++ b/drivers/video/tegra/host/nvdla/nvdla_ioctl.c
@@ -1329,7 +1329,7 @@ static int nvdla_release(struct inode *inode, struct file *file)
 
 const struct file_operations tegra_nvdla_ctrl_ops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = nvdla_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = nvdla_ioctl,
diff --git a/drivers/video/tegra/host/pva/pva_ioctl.c b/drivers/video/tegra/host/pva/pva_ioctl.c
index 62f1f75c..4c2c85a4 100644
--- a/drivers/video/tegra/host/pva/pva_ioctl.c
+++ b/drivers/video/tegra/host/pva/pva_ioctl.c
@@ -1134,7 +1134,7 @@ static ssize_t pva_read_vpu_print_buffer(struct file *file,
 
 const struct file_operations tegra_pva_ctrl_ops = {
 	.owner = THIS_MODULE,
-	.llseek = no_llseek,
+	.llseek = NULL,
 	.unlocked_ioctl = pva_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = pva_ioctl,
diff --git a/drivers/virt/tegra/tegra_hv_pm_ctl.c b/drivers/virt/tegra/tegra_hv_pm_ctl.c
index 6e493c12..b70d1302 100644
--- a/drivers/virt/tegra/tegra_hv_pm_ctl.c
+++ b/drivers/virt/tegra/tegra_hv_pm_ctl.c
@@ -389,7 +389,7 @@ static int tegra_hv_pm_ctl_release(struct inode *inode, struct file *filp)
 
 static const struct file_operations tegra_hv_pm_ctl_fops = {
 	.owner		= THIS_MODULE,
-	.llseek		= no_llseek,
+	.llseek		= NULL,
 	.read		= tegra_hv_pm_ctl_read,
 	.write		= tegra_hv_pm_ctl_write,
 	.poll		= tegra_hv_pm_ctl_poll,
-- 
2.34.1

